src/librustc_trans/mir/rvalue.rs

```diff
@@ -114,7 +114,9 @@ impl<'a, 'tcx> FunctionCx<'a, 'tcx> {
                     }
 
                     // Use llvm.memset.p0i8.* to initialize byte arrays
-                    let v = base::from_immediate(&bx, v);
+                    if common::val_ty(v) == Type::i1(bx.cx) {
+                        v = bx.zext(v, Type::i8(bx.cx))
+                    }
```

Comment:
Am not sure about this, but I feel like boolean arrays should also use `memset` when they can, right?

Reply:
Indeed.